### ISSUE TITLE
Properly handle collisions between message ids in txpool

### ISSUE THREAD

**Issue by Voxelot** [FuelLabs/fuel-core#511]
This work was deferred from the main task (#510) of upgrading to the latest versions of fuel-tx and fuel-vm, as it is likely blocked on #485 as well to check if the message exists in the database.

Test Cases:
- [x] tx is successfully inserted when input message id exists in db
- [x] tx is rejected from the txpool if input message id does not exist in db
- [x] tx is rejected from the txpool if gas_price is lower than another transaction with the same message id input
- [x] a successfully inserted tx is squeezed out when second tx using the same message id input is inserted with a higher gas price
